include/effects/matrix/PatternSMSupernova.h

```diff
@@ -4,86 +4,96 @@
 
 // Inspired by https://editor.soulmatelights.com/gallery/1923-supernova
 
-class PatternSMSupernova : public LEDStripEffect 
-{
-    void Init()
-    {
-        memset(enlargedObjectTime, 0, sizeof(enlargedObjectTime));
-        memset(PosX, 0, sizeof(PosX));
-        memset(PosY, 0, sizeof(PosY));
-        memset(Hue, 0, sizeof(Hue));
-        memset(State, 0, sizeof(State));
-        memset(SpeedX, 0, sizeof(SpeedX));
-        memset(SpeedY, 0, sizeof(SpeedY));
-        memset(IsShift, 0, sizeof(IsShift));
-    }
 
+class PatternSMSupernova : public LEDStripEffect
+{
 public:
 
-    PatternSMSupernova() : LEDStripEffect(EFFECT_MATRIX_SMSUPERNOVA, "Supernova"), hue(0), hue2(0), step(0), deltaValue(0), enlargedObjectNUM(0) 
+    PatternSMSupernova() : LEDStripEffect(EFFECT_MATRIX_SMSUPERNOVA, "Supernova"), hue(0), hue2(0), step(0), enlargedDebrisItemNUM(0)
     {
     }
 
-    PatternSMSupernova(const JsonObjectConst &jsonObject) : LEDStripEffect(jsonObject) 
+    PatternSMSupernova(const JsonObjectConst &jsonDebrisItem) : LEDStripEffect(jsonDebrisItem)
     {
     }
 
-    virtual size_t DesiredFramesPerSecond() const override 
+    virtual size_t DesiredFramesPerSecond() const override
     {
-        return 40;
+        return 60;
     }
 
-    void Start() override 
+    void Start() override
     {
-        Init();
         g()->Clear();
-        enlargedObjectNUM = 200;
-        if (enlargedObjectNUM > MAX_COUNT)
-            enlargedObjectNUM = 200U;
-        deltaValue = enlargedObjectNUM / (sqrt3(CENTER_X_MAJOR * CENTER_X_MAJOR + CENTER_Y_MAJOR * CENTER_Y_MAJOR) * 4U) + 1U;
-        for (unsigned i = 0; i < enlargedObjectNUM; i++)
-            IsShift[i] = false;
+        enlargedDebrisItemNUM = 200;
+        if (enlargedDebrisItemNUM > MAX_COUNT)
+            enlargedDebrisItemNUM = 200U;
     }
 
     void Draw() override {
         step = -1;
         g()->DimAll(200);
-        for (unsigned i = 0; i < enlargedObjectNUM; i++) {
-            if (!IsShift[i] && step) {
-                starfield2Emit(i);
+
+        for (auto& debris_item : _debris_items) {
+            if (!debris_item._is_shift && step) {
+                StarfieldEmit(debris_item);
```

Comment:
(etc.) Shall I just ignore the bracing beyond what you changed in your most recent commit? :)